lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -45,9 +57,98 @@
     seed_milestone_relationship_types,
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test if user is coach on ccx.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up course for tests
+        """
+        super(CoachAccessTestCaseCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CoachAccessTestCaseCCX, self).setUp()
+
+        # Create ccx coach account
+        self.coach = AdminFactory.create()
+        self.client.login(username=self.coach.username, password="test")
+
+        # assign role to coach
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(self.coach)
```

Comment:
I made this comment because it doesn't appear that `role` *is* saved to the database. You are creating an instance of a `CourseCcxCoachRole`, and you are adding users to it, but you never call `.save()` on that instance. As a result, it won't be persisted to the database, and will vanish as soon as the `role` variable falls out of scope, i.e. at the end of this `setUp()` method.